Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -271,6 +280,23 @@ private void OnBallisticAmmoCount(EntityUid uid, BallisticAmmoProviderComponent
         args.Capacity = component.Capacity;
     }
 
+    /// <summary>
+    /// Adds one ammo to the ammo provider entity.
+    /// This assumes the ammo already passed the whitelist check from <see cref="BallisticAmmoProviderComponent.Whitelist">
+    /// </summary>
+    private void FillAmmo(Entity<BallisticAmmoProviderComponent> ammoProvider, EntityUid ammo, InteractUsingEvent args)
```

Comment:
When passing an event around it should always be by `ref`. However it's also rare to need to pass the event around, and not needed here.

The `args` parameter is used twice in this function. One is for `args.Target`, which is a redundant variable because it's always the same as entity it was raise on, i.e. the entity we already have `ammoProvider`.

That leaves `args.User` as the only part we use from `args`, so the param can be simplified to that variable type `EntityUid`. It can then be made even more specific, because `PlayPredicted()` takes an `EntityUid?` for the user.
```suggestion
    private void FillAmmo(Entity<BallisticAmmoProviderComponent> ammoProvider, EntityUid ammo, EntityUid? user)
```